query/src/test/java/tech/ydb/query/impl/QueryIntegrationTest.java

```diff
@@ -0,0 +1,359 @@
+        Assert.assertEquals(id, s2.getId());
+        s2.updateSessionState(Status.of(StatusCode.ABORTED));
+        s2.close();
+
+        SessionImpl s3 = (SessionImpl)queryClient.createSession(Duration.ofSeconds(5)).join().getValue();
+        Assert.assertEquals(id, s3.getId());
+        s3.updateSessionState(Status.of(StatusCode.BAD_SESSION));
+        s3.close();
+
+        SessionImpl s4 = (SessionImpl)queryClient.createSession(Duration.ofSeconds(5)).join().getValue();
+        Assert.assertNotEquals(id, s4.getId());
+        s4.close();
+    }
+
+    @Test
+    public void testSimpleCRUD() {
+        List<Entity> entities = new ArrayList<>();
+        entities.add(new Entity(1, "entity 1", BYTES_EMPTY, true));
+        entities.add(new Entity(2, "entity 2", BYTES_EMPTY, true));
+        entities.add(new Entity(3, "entity 3", BYTES_LEN2, false));
+        entities.add(new Entity(3, "dublicate", BYTES_LEN5, true));
+        entities.add(new Entity(5, "entity 5", BYTES_LEN2, false));
+
+        for (Entity entity: entities) {
+            String query = "UPSERT INTO `" + TEST_TABLE + "` (id, name, payload, is_valid) "
+                    + "VALUES ($id, $name, $payload, $is_valid)";
+
+            Params params = Params.of(
+                    "$id", PrimitiveValue.newInt32(entity.id),
+                    "$name", PrimitiveValue.newText(entity.name),
+                    "$payload", PrimitiveValue.newBytes(entity.payload),
+                    "$is_valid", PrimitiveValue.newBool(entity.isValid)
+            );
+
+            try (QuerySession session = queryClient.createSession(SESSION_TIMEOUT).join().getValue()) {
+                session.createQuery(query, QueryTx.SERIALIZABLE_RW, params)
+                        .execute(this::printQuerySetPart)
+                        .join().getStatus().expectSuccess();
+            }
+        }
+
+        try (QuerySession session = queryClient.createSession(Duration.ofSeconds(5)).join().getValue()) {
+            String query = "SELECT id, name, payload, is_valid FROM " + TEST_TABLE + " ORDER BY id;";
+            session.createQuery(query, QueryTx.SERIALIZABLE_RW)
+                    .execute(this::printQuerySetPart)
+                    .join().getStatus().expectSuccess();
+        }
+
+        try (QuerySession session = queryClient.createSession(SESSION_TIMEOUT).join().getValue()) {
+            session.createQuery("DELETE FROM " + TEST_TABLE, QueryTx.SERIALIZABLE_RW)
+                    .execute(this::printQuerySetPart)
+                    .join().getStatus().expectSuccess();
+        }
+    }
+
+    public void printQuerySetPart(QueryResultPart part) {
+        ResultSetReader rs = part.getResultSetReader();
+        if (rs != null) {
+            logger.info("got query result part with index {} and {} rows", part.getResultSetIndex(), rs.getRowCount());
+        }
+    }
+
+    @Test
+    @Ignore
```

Comment:
Is the test not finished yet?

Reply:
It is special test case for https://github.com/ydb-platform/ydb/issues/2699. Now this issue is fixed in main branch of YDB, but is not available in docker image